.github/workflows/lint.yml

```diff
@@ -0,0 +1,44 @@
+name: Rust Lint
+
+on:
+  pull_request:
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  check_format_and_lint:
+    continue-on-error: true # we want to show this check failure in GH but to not prevent building.
+    name: Assert Format and Run Clippy
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v2
+
+      - name: Install Rust
+        uses: actions-rs/toolchain@v1
+        id: toolchain
+        with:
+          toolchain: nightly
+          profile: minimal
+          components: rustfmt, clippy
+
+      ### BUILD CACHE ###
+      - name: Cache Cargo registry, target, index
```

Comment:
Are the caching additions you've made elsewhere relevant here as well?
i.e.
```
id: cache-cargo
env:
  cache-name: cache-cargo
```

Reply:
Answer to myself: in this commit, maybe so. But this is DRYed out in the following commit.

Reply:
Follow up, post-revert: this comment may be relevant again!

Reply:
And hopefully full circle, it looks like they've been added. What a ride!